### PR TITLE
flexibility improvements for signing transactions

### PR DESCRIPTION
**Pull request by piercetrey-figure**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -67,6 +67,10 @@ subprojects {
     configurations.forEach { it.exclude("org.slf4j", "slf4j-api") }
     val artifactName = "pb-grpc-$name-kotlin"
 
+    tasks.withType<PublishToMavenLocal> {
```
Comment (piercetrey-figure) [inline build.gradle.kts:70]: allows publishing locally without having to add `-x signMavenPublication`

File: client-common/src/main/kotlin/io/provenance/client/grpc/BaseReq.kt [hunk 2]
```diff
@@ -34,7 +39,12 @@ data class BaseReqSigner(
                     .setSingle(Single.newBuilder().setModeValue(SignMode.SIGN_MODE_DIRECT_VALUE))
             )
             .setSequence(this.account!!.sequence + this.sequenceOffset)
-            .build()
+            .build().let {
+                customizeSignerInfo?.invoke(it) ?: it
```
Comment (piercetrey-figure) [inline client-common/src/main/kotlin/io/provenance/client/grpc/BaseReq.kt:43]: Needed for flexibility on modifying the signerInfo doc by the client

File: client-coroutines/src/main/kotlin/io/provenance/client/coroutines/PbCoroutinesClient.kt [hunk 3]
```diff
@@ -204,12 +204,13 @@ open class PbCoroutinesClient(
...
  * @param bech32Address The bech32 address to fetch.
- * @return [cosmos.auth.v1beta1.Auth.BaseAccount] or throw [IllegalArgumentException] if the account type is not supported.
+ * @return [BaseAccount] or throw [IllegalArgumentException] if the account type is not supported.
  */
-suspend fun QueryGrpcKt.QueryCoroutineStub.getBaseAccount(bech32Address: String): cosmos.auth.v1beta1.Auth.BaseAccount =
+suspend fun QueryGrpcKt.QueryCoroutineStub.getBaseAccount(bech32Address: String, unpackAccount: (Any.() -> BaseAccount)? = null): BaseAccount =
     account(QueryOuterClass.QueryAccountRequest.newBuilder().setAddress(bech32Address).build()).account.run {
         when {
-            this.`is`(cosmos.auth.v1beta1.Auth.BaseAccount::class.java) -> unpack(cosmos.auth.v1beta1.Auth.BaseAccount::class.java)
+            unpackAccount != null -> unpackAccount()
```
Comment (piercetrey-figure) [inline client-coroutines/src/main/kotlin/io/provenance/client/coroutines/PbCoroutinesClient.kt:212]: allows for handling special types of accounts by the client

File: client-coroutines/src/main/kotlin/io/provenance/client/coroutines/PbCoroutinesClient.kt [hunk 4]
```diff
@@ -173,7 +173,7 @@ open class PbCoroutinesClient(
             .setMode(actualMode)
             .build()
         ).let { res ->
-            if (simulateBlock) {
+            if (simulateBlock && res.txResponse.code == 0) {
```
Comment (piercetrey-figure) [inline client-coroutines/src/main/kotlin/io/provenance/client/coroutines/PbCoroutinesClient.kt:176]: noticed that a broadcast with a checktx failure would poll until the timeout 😬

**Review by mtps — APPROVED** [review]

**Pull request merged**